src/tox_recreate/plugin.py

```diff
@@ -0,0 +1,57 @@
+from functools import lru_cache
+from hashlib import sha512
+from pathlib import Path
+
+import pluggy
+
+hookimpl = pluggy.HookimplMarker("tox")
+
+
+@lru_cache
+def cached_hash_path(envconfig):
+    """Return the path to envconfig's cached hash file."""
+    return Path(envconfig.envdir) / "tox_recreate.hash"
+
+
+@lru_cache
+def cached_hash(envconfig):
+    """Return envconfig's cached hash."""
+    try:
+        return cached_hash_path(envconfig).read_text()
+    except FileNotFoundError:
+        return None
+
+
+def get_setup_cfg_path():  # pragma: no cover
+    """Return the path to the project's setup.cfg file."""
+    return Path("setup.cfg")
```

Comment:
This is pulled out into a separate function so that the tests can patch it in order to make the code read a test `setup.cfg` file instead of the project's real one.